src/AssetsTransferApi.ts

```diff
@@ -234,10 +265,39 @@ export class AssetsTransferApi {
 			xcmDirection === Direction.SystemToSystem &&
 			!containsNativeRelayAsset(assetIds, nativeRelayChainAsset);
 
-		if (assetType === AssetType.Foreign || isSystemToSystemReserveTransfer) {
+		// if transfering a foreign asset, we determine if the asset is the native asset
+		// of the dest parachain, if so we construct a teleport
+		const isSystemToParaTeleport =
```

Comment:
What do you think if we put all this logic in `isSystemToParaTeleport`, and `isSystemToParaReserveTransfer` inside of `checkAssetType`. Then we can change the return `enum` to be something like 

```
export enum AssetCallType {
	Reserve = 'Reserve',
	Teleport = 'Teleport',
}
```

That we things can be a little more abstracted, and have a single source of truth.

Reply:
Then the If statement should be as simple as `if (assetCallType === AssetType.Reserve)`